File: 2019-01-13/1003/1003.java
import java.io.*;

class Main{
	static int []arr = new int[50];
	static int []one = new int[50];
	static int []zero = new int[50]; 

	public static void main(String[] args) throws Exception{
		BufferedReader bfr = new BufferedReader(new InputStreamReader(System.in));
		int T = Integer.parseInt(bfr.readLine());
		one[0] = 0; one[1] = 1;
		zero[0] = 1; zero[1] = 0;
		for(int i = 0; i < T; i++){
			int N = Integer.parseInt(bfr.readLine());
			fibo(N);
			System.out.printf("%d %d\n", zero[N], one[N]);
		}
	}

	public static int fibo(int N){
		if(N == 0 || N == 1)
			return N;
		else {
			if(arr[N] == 0){
				arr[N] = fibo(N-1) + fibo(N-2);
				one[N] = one[N-1] + one[N-2];
				zero[N] = zero[N-1] + zero[N-2];
			}
			return arr[N];
		}
	}
}

File: 2019-01-13/1003/1003.c
#include <stdio.h>
#include <string.h>

int arr[46]= {0,};
int zeroArr[46] = {1, 0};
int oneArr[46] = {0, 1};

int fibo(int n){
	if (n == 0) {
        return 0;
    } else if (n == 1) {
        return 1;
    }else{
    	if(arr[n] == 0){
    		arr[n] = fibo(n-1) + fibo(n-2);
    		zeroArr[n] = zeroArr[n-1] + zeroArr[n-2];
    		oneArr[n] = oneArr[n-1] + oneArr[n-2];
    	}
    	return arr[n];
    }
}

int main(T){
	int N;
	scanf("%d", &T);
	for(int i =0; i < T; i++){
		scanf("%d", &N);
		fibo(N);
		printf("%d %d\n", zeroArr[N], oneArr[N]);
	}
}